great_expectations/datasource/types/batch_kwargs.py

```diff
@@ -0,0 +1,144 @@
+from hashlib import md5
+import datetime
+
+import pandas as pd
+from six import string_types
+from great_expectations.types import RequiredKeysDotDict, ClassConfig
+
+try:
+    import pyspark
+except ImportError:
+    pyspark = None
+
+
+class BatchKwargs(RequiredKeysDotDict):
+    """BatchKwargs represent uniquely identifying information for a Batch of data.
+
+    BatchKwargs are generated by BatchGenerators and are interpreted by datasources.
+
+    Note that by default, the partition_id *will* be included both in the partition_id portion and in the batch_kwargs
+    hash portion of the batch_id.
+
+    """
+    _required_keys = set()
+    # FIXME: Need discussion about whether we want to explicitly ignore, explicitly include, or just use whatever
+    # is present
+
+    _partition_id_key = "partition_id"
+    _batch_id_ignored_keys = {
```

Comment:
Right. I'll add better documentation. The idea is you can include keys here that we want to definitively exclude from the batch_id (so that two batches with different values here will have the same id)